core/src/replay_stage.rs

```diff
@@ -997,7 +999,17 @@ impl ReplayStage {
         transaction_status_sender: Option<TransactionStatusSender>,
         replay_vote_sender: &ReplayVoteSender,
         verify_recyclers: &VerifyRecyclers,
+        subscriptions: Option<Arc<RpcSubscriptions>>,
     ) -> result::Result<usize, BlockstoreProcessorError> {
+        if bank_progress.replay_progress.num_entries == 0 {
```

Comment:
hmm this may not be exactly what you want, since it's possible here that `num_entries ==0` both before and after this function call.  

If you want the time when the bank is actually created in `generate_new_bank_forks()`, then at the line `let child_bank = Self::new_bank_from_parent_with_notify(` for validators, and in `maybe_start_leader`, `Self::new_bank_from_parent_with_notify(` for leaders will be the most straightforward places to put the notification

Reply:
Also of note, the only time this will be dramatically different than the first data shred received should be if replay is behind, i.e. still replaying the previous slot/different fork when the next slot's data shred has already arrived.

Reply:
Ah good point. My intent was to get a timestamp for when a node actually started replaying the first entry of a block which I think I could do by moving this into `confirm_slot` and checking if entries isn't empty. Does that feel useful?

Regardless, sending a notification for when the bank is created seems more useful since it indicates when a node decided to build off another fork

Reply:
> Also of note, the only time this will be dramatically different than the first data shred received should be if replay is behind, i.e. still replaying the previous slot/different fork when the next slot's data shred has already arrived.

Well, my hope was to capture any issues with turbine here where replay got delayed due to slow shred propagation. Sounds like the only way to do that is to capture whether or not the node is behind with a "block start" notification.

Reply:
Replaced this with a `CreatedBank` notification